### PR TITLE
Modified reports to generate a monthly, by person report.

### PR DESCRIPTION
**Pull request by jason-at-scroll**

### CODE REVIEW

File: gitleaves/reports.py [hunk 1]
```diff
@@ -68,16 +72,29 @@ def get_next_leaves_by_month(bydates):
...
+#                data['bymonths'][month].items())
+#     csv.writer(open(outfile, 'w')).writerows(outdata)
+#     print(f'Data exported to {outfile}')
+
+
+def export_month_csv(some_group_of_people):
+    people_by_month = some_group_of_people['group_of_people_by_month']
+    template = templates.get_template('ghwiki/Monthwise.md')
+    if not os.path.exists(ghwiki_reports_dir):
+        os.makedirs(ghwiki_reports_dir)
```
Comment (shon) [inline gitleaves/reports.py]: @jason-at-scroll  Code to ensure ghwiki_reports_dir exists is repeated a few time in the module. I suggest moving this check just after the paths declaration (currently line 21).

File: gitleaves/reports.py [hunk 2]
```diff
@@ -86,12 +103,17 @@ def gen_ghwiki_reports():
...
+    export_month_csv(data)
+
     template = templates.get_template('ghwiki/Home.md')
     if not os.path.exists(ghwiki_reports_dir):
         os.makedirs(ghwiki_reports_dir)
     with open(f'{ghwiki_reports_dir}/Home.md', 'w') as report:
         report.write(template.render(today_leaves=today_leaves,
                                      next_leaves_by_month=next_leaves_by_month))
+
+    place_reports()
```
Comment (shon) [inline gitleaves/reports.py]: @jason-at-scroll 
- for now drop this new function `place_reports` just call copyfile here 
- not a very strong opinion but open(dst/__Sidebar.md, 'w').write(src/_Sidebar.md) might be slightly better than importing shutils.copyfile

File: gitleaves/reports.py [hunk 3]
```diff
@@ -68,16 +72,29 @@ def get_next_leaves_by_month(bydates):
...
+# Shekhar’s original in case Jason messes up, — mjb 12/20/2021
+# def export_month_csv(year, month):
+#     leaves_csv_path = leaves_csv_path_pat.format(YYYY=year)
+#     data = load_csv(leaves_csv_path)
+#     outfile = f'/tmp/leaves.{year}.{month}.csv'
+#     outdata = ((d.isoformat(), *names)
+#                for d, names in
+#                data['bymonths'][month].items())
+#     csv.writer(open(outfile, 'w')).writerows(outdata)
+#     print(f'Data exported to {outfile}')
```
Comment (shon) [inline gitleaves/reports.py]: I understood why you have chosen to do keep commented out code but it is not needed. 
Git does this for us. Infact this has hindered the diff view

File: gitleaves/reports.py [hunk 4]
```diff
@@ -68,16 +72,29 @@ def get_next_leaves_by_month(bydates):
...
+
+
+def export_month_csv(some_group_of_people):
+    people_by_month = some_group_of_people['group_of_people_by_month']
```
Comment (shon) [inline gitleaves/reports.py]: Naming suggestions
- some_group_of_people -> data
- group_of_people_by_month -> bymonthnames_groupedby_applicants / bymonthnames_byapplicants

File: gitleaves/reports.py [hunk 5]
```diff
@@ -86,12 +103,17 @@ def gen_ghwiki_reports():
     next_leaves_by_month = ((calendar.month_name[month], leaves)
                             for month, leaves
                             in next_leaves_by_month_temp_jar.items())
+    export_month_csv(data)
```
Comment (shon) [inline gitleaves/reports.py]: For now: Drop this function `export_month_csv` just move the function code here

File: gitleaves/templates/ghwiki/Monthwise.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{% for month, people in people_by_month.items() %}
+
+## {{ month }}  
+
+{% for name, days in people.items() %}  
+**{{ name }} :**  
+
+{%- for day in days %}  
+{{ day.strftime('%Y-%m-%d') }}
```
Comment (shon) [inline gitleaves/templates/ghwiki/Monthwise.md]: How about `### {{ name }}` vs `**{{ name }}**`

Also Ordered list might be better for listing days

`1. {{ day.strftime('%Y-%m-%d') }} `

File: gitleaves/__init__.py [hunk 7]
```diff
@@ -2,4 +2,4 @@
 
 __author__ = """Shekhar Tiwatne"""
 __email__ = 'pythonic@gmail.com'
-__version__ = '0.4.7'
+__version__ = '0.4.8'
```
Comment (shon) [inline gitleaves/__init__.py]: Let's call it 0.5.0 as this is not a patch but a new minor release with a new report.

**Review by shon — CHANGES_REQUESTED** [review]

File: gitleaves/reports.py (hunk 3 above)
Comment (jason-at-scroll) [reply gitleaves/reports.py]: Shoot! this was just temporary while i worked on this. It *was* supposed to come out before the pr 😂
Removed.

File: gitleaves/reports.py (hunk 4 above)
Comment (jason-at-scroll) [reply gitleaves/reports.py]: lessons learnt
1. Make function parameters as generic/abstract as possible
2. Make variables as descriptively precise as you can possibly make them

Changes done.

File: gitleaves/templates/ghwiki/Monthwise.md (hunk 6 above)
Comment (jason-at-scroll) [reply gitleaves/templates/ghwiki/Monthwise.md]: - Did not think of name been a structurally different thing from the dates. Makes sense! Changed to h3
- I thought about doing a list, then second guessed myself :) Dates are now lists.

File: gitleaves/__init__.py (hunk 7 above)
Comment (jason-at-scroll) [reply gitleaves/__init__.py]: This is something I need to sit with you and understand ( possibly need multiple rounds of feedback on.)
Terribly confused over what constitutes what.

Changed to 0.5

File: gitleaves/reports.py (hunk 1 above)
Comment (jason-at-scroll) [reply gitleaves/reports.py]: Homer Simpson - - Homer Simpson Doh Png @clipartmax.com

Comment (jason-at-scroll) [reply gitleaves/reports.py]: I need to be more proactive and less apprehensive about changes like this.

File: gitleaves/reports.py (hunk 5 above)
Comment (jason-at-scroll) [reply gitleaves/reports.py]: Done

File: gitleaves/reports.py (hunk 2 above)
Comment (jason-at-scroll) [reply gitleaves/reports.py]: Done!

**Pull request merged**